gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -186,42 +186,11 @@ private ManagedChannel createSingleChannel() throws IOException {
 
     // TODO(hzyi): Change to ManagedChannelBuilder directly when
     // https://github.com/grpc/grpc-java/issues/4050 is resolved.
-    ManagedChannelBuilder builder;
+    ManagedChannelBuilder builder = GoogleDefaultChannelBuilder.forAddress(serviceAddress, port);
     if (maxInboundMetadataSize != null) {
-      Class<?> nettyChannelBuilderClass;
-      try {
-        nettyChannelBuilderClass =
-            Class.forName("io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder");
-      } catch (ClassNotFoundException e) {
-        try {
-          nettyChannelBuilderClass = Class.forName("io.grpc.netty.NettyChannelBuilder");
-        } catch (ClassNotFoundException ex) {
-          throw new RuntimeException(
-              "Unable to create the channel because neither"
-                  + " \"io.grpc.netty.NettyChannelBuilder\" nor"
-                  + " \"io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder\" is found.");
-        }
-      }
-      try {
-        Object object =
-            nettyChannelBuilderClass
-                .getMethod("forAddress", String.class, int.class)
-                .invoke(null, serviceAddress, port);
-        object =
-            nettyChannelBuilderClass
-                .getMethod("maxHeaderListSize", int.class)
-                .invoke(object, maxInboundMetadataSize);
-        builder = (ManagedChannelBuilder) object;
-      } catch (NoSuchMethodException
-          | IllegalAccessException
-          | IllegalArgumentException
-          | InvocationTargetException e) {
-        throw new RuntimeException(
-            "Unable to set maxHeaderListSize due to exception: " + e.getMessage());
-      }
-    } else {
-      builder = ManagedChannelBuilder.forAddress(serviceAddress, port);
+      builder.maxInboundMetadataSize(maxInboundMetadataSize);
     }
+
     builder
```

Comment:
I feel you could move these `intercept`s up before `maxInboundMetadataSize` since these are required while `maxInboundMetadataSize` is optional.